nomad/host_volume_endpoint_test.go

```diff
@@ -822,6 +1009,12 @@ func (v *mockHostVolumeClient) setDelete(errMsg string) {
 func (v *mockHostVolumeClient) Create(
 	req *cstructs.ClientHostVolumeCreateRequest,
 	resp *cstructs.ClientHostVolumeCreateResponse) error {
+
+	// block until something runs unblockCurrent()
+	if bc := v.getBlockChan(); bc != nil {
+		bc <- "create"
+	}
```

Comment:
I refactored it (again), and while it's still convoluted, it does actually test the serialization.

instead of these mock client methods blocking on a _receive_, they block on a _send_. then `unblockCurrent()` unblocks them by receiving, and it returns what it got, i.e. whichever client RPC was being called at the time.

then the test code makes sure that matches which server RPC completed, and we can check the state as appropriate for the operation type.